packages/x-data-grid/src/models/api/gridCallbackDetails.ts

```diff
@@ -8,4 +9,8 @@ export interface GridCallbackDetails<K extends keyof GridControlledStateReasonLo
    * The reason for this callback to have been called.
    */
   reason?: GridControlledStateReasonLookup[K];
+  /**
+   * GridApi that let you manipulate the grid.
+   */
+  api?: GridApiCommunity;
```

Comment:
Not sure if we should add the typing or remove the field. Also I don't love too much having optional fields, it doesn't seem very clear for an external user if they should expect an API object or not. I've tagged the rest of the team for review, let's see what they think.

Reply:
I think we should make the `api` required.
We used to pass it conditionally because `apiRef` was a Pro feature back then.
It's no longer the case – the `apiRef` is available in every plan.

```diff
diff --git a/packages/x-data-grid/src/hooks/core/useGridStateInitialization.ts b/packages/x-data-grid/src/hooks/core/useGridStateInitialization.ts
index 860d4d505..0ab86b644 100644
--- a/packages/x-data-grid/src/hooks/core/useGridStateInitialization.ts
+++ b/packages/x-data-grid/src/hooks/core/useGridStateInitialization.ts
@@ -92,11 +92,7 @@ export const useGridStateInitialization = <PrivateApi extends GridPrivateApiComm
         const model = controlState.stateSelector(newState, apiRef.current.instanceId);
 
         if (controlState.propOnChange && hasPropChanged) {
-          const details =
-            props.signature === GridSignature.DataGridPro
-              ? { api: apiRef.current, reason }
-              : { reason };
-          controlState.propOnChange(model, details);
+          controlState.propOnChange(model, { api: apiRef.current, reason });
         }
```